File: StackOverflow/pipes/read.c
#include "stdio.h"
#include "stdlib.h"

int main(int argc, char** ARGV) {
   char buffer[1024];
   while(1) {
      if (fgets(buffer, sizeof(buffer), stdin) != NULL) {
         printf("%s", buffer);
      } else {
         printf("eof\n");
      }
      sleep(1);
   }
   return 0;
}

File: StackOverflow/pipes/read.pl
use strict;
use warnings;

while (1) {
    seek *STDIN, 0, 0;
    print "eof: ".eof(*STDIN)."\n";
    if (!eof(*STDIN)) {
        print "Cleared!";
        while(<STDIN>) {}
        print "eof: ".eof(*STDIN)."\n";
    }
    sleep 2;
}
